pages/index.tsx

```diff
@@ -30,9 +32,15 @@ export default function Home({
 
   const [showUserWindow, setShowUserWindow] = useState(false);
 
-  const { isFetching, isError, error } = useGetGifts();
-
-  useShowErrorToast(error);
+  const createGiftQuery = useMutation({
+    mutationKey: QueryKeys.CREATE_GIFT,
+    mutationFn: async (newGift: CreateGift) => await createGift(newGift),
+    // if success: refresh giftlist
+    onSuccess: async () =>
+      await queryClient.invalidateQueries({
+        queryKey: QueryKeys.GIFTS,
+      }),
+  });
```

Comment:
Olisiko tässä tarvetta vaihtaa 
```suggestion
  const { mutateAsync, isPending } = useMutation({
    mutationKey: QueryKeys.CREATE_GIFT,
    mutationFn: async (newGift: CreateGift) => await createGift(newGift),
    // if success: refresh giftlist
    onSuccess: async () =>
      await queryClient.invalidateQueries({
        queryKey: QueryKeys.GIFTS,
      }),
  });
```

En tehnyt kyseistä muokkausta vielä, koska tähän asti esim. `isPending` on liittynyt aina `useGetGifts()`-funktioon

Reply:
En ehkä tekisi tuota muokkausta. Muuten on vähän epäselvää mihin tietty `isPending` liittyy, kun taas `createGiftQuery.isPending` näkee heti mihin se liittyy.

Reply:
Joo, samaa mieltä